spec.go

```diff
@@ -11,4 +11,8 @@ type Handler interface {
 	// source code. Inspecting an masked error shows where the error was passed
 	// through within the code base. This is gold for debugging and bug hunting.
 	Mask(err error) error
+	// Maskf is like Mask. In addition to that it takes a format string and
+	// variadic arguments like fmt.Sprintf. The format string and variadic
+	// arguments are used to annotate the given error.
+	Maskf(err error, f string, v ...interface{}) error
```

Comment:
The reason why I dropped `Maskf` in my proposal was because I think it gets abused. I think an error should be annotated with additional information about the root cause. In practise people use `Maskf` everywhere to append information they think are important for all kinds of reasons where the error is forwarded to the caller. `Maskf` is used in places where the error does not originate from and I think this is wrong. When you look at error stacks to debug a situation you are only interested in the error message and the stack. The stack is unrelated to the message. With providing `Maskf` anybody can change the annotation of the error everywhere, and I think this is wrong and misleading. With this we will not even be aware of the original annotation being overwritten somewhere. The proposal I made ensured that you cannot easily mislead yourself. `SetDescf` from my proposal would only be usable at the root cause of a `microerror` error. When you want to use `microerror.Maskf` instead of `*Error.SetDescf`, `Maskf` should only accept `*Error` instead of `error`. This would then resemble the same level of confidence as in my original proposal and somewhat guarantee to always only have the desired annotation present. Important to me is to not annotate in the middle of an error stack, but only at its root.

Reply:
So now `Maskf` takes `*Error`. This also implied changes in `DeprecatedSetStack`. I changed it to `DeprecatedPrependStack`.